worlds/AutoWorld.py

```diff
@@ -378,6 +378,19 @@ def pre_fill(self) -> None:
         """Optional method that is supposed to be used for special fill stages. This is run *after* plando."""
         pass
 
+    def modify_early_locations(self, early_locations: Dict[int, List["Location"]]) -> None:
+        """
+        Gets called as part of distribute_early_items and early items plando.
+        Can be used to modify which locations are considered early.
+        """
+        # make a clean copy of state with just start inventory
+        sweep_state = self.multiworld.state.copy()
+        # collect any events from our already reachable locations
+        for location in early_locations[self.player]:
+            if location.is_event and location.advancement:
+                sweep_state.collect(location.item, event=True, location=location)
+        early_locations[self.player] += self.multiworld.get_reachable_locations(sweep_state, self.player)
```

Comment:
Would this not duplicate locations in the early_locations list if they're not events and are reachable with starting inventory? They'd get added in `get_early_locations`, passed, in an never removed.

Also unsure about passing in the entire `early_locations` dict instead of just the list pertaining to the player. It invites bugs if someone ever uses the wrong index for some reason, and I don't know why a world dev would ever need to access the early locations list for other players, especially when you don't know which lists have already been modified.

Recommendations:
* Pass in `early_locations` as just a `List['Location']` pertaining only to the world's player
  * Things can still be cleared with `list.clear()` / `list.remove()` / etc
* Use `early_locations.clear()` and `early_locations.extend(self.multiworld.get_reachable_locations())` instead of `+=`